programs/bench.c

```diff
@@ -617,43 +683,50 @@ static void BMK_benchFileTable(const char* const * const fileNamesTable, unsigne
     if (!srcBuffer) EXM_THROW(12, "not enough memory");
 
     /* Load input buffer */
-    BMK_loadFiles(srcBuffer, benchedSize, fileSizes, fileNamesTable, nbFiles);
+    BMK_loadFiles(srcBuffer, benchedSize, fileSizes, fileNamesTable, nbFiles, printable);
 
     /* Bench */
     if (g_separateFiles) {
         const BYTE* srcPtr = (const BYTE*)srcBuffer;
         U32 fileNb;
+        BMK_result_t* resultarray = (BMK_result_t*)malloc(sizeof(BMK_result_t) * nbFiles);
+        if(resultarray == NULL) EXM_THROW(12, "not enough memory");
         for (fileNb=0; fileNb<nbFiles; fileNb++) {
             size_t const fileSize = fileSizes[fileNb];
-            BMK_benchCLevel(srcPtr, fileSize,
+            resultarray[fileNb] = BMK_benchCLevel(srcPtr, fileSize,
                             fileNamesTable[fileNb], cLevel, cLevelLast,
-                            fileSizes+fileNb, 1,
-                            dictBuffer, dictBufferSize, compressionParams);
+                            fileSizes+fileNb, 1, dictBuffer, dictBufferSize, 
+                            compressionParams,printable);
             srcPtr += fileSize;
         }
+        returnval =  BMK_average(resultarray, nbFiles, fileSizes);
```

Comment:
It's a bit more useful to generate an average result over multiple files (rather than multiple compression levels)
but there are 2 problems : 
- The formula in `BMK_average()` is wrong. Speeds don't add up, time / duration would be more sensible.
- This functionality is already integrated in `benchMem()`.
  It's actually the default behavior.
  This function is actually a hack around that, to prevent `benchMem()` from generating an aggregate result,
   and have one clean result per file instead, in order to precisely observe the differences.

So, here also, I believe it's not a correct entry point for `paramgrill`.
Skip that stage, create an intermediate one, that fits `paramgrill` needs, and also fits nicely with the other benchmark functions, which are created for other needs.